File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/codebase/find/CodeBasesListInventory.java
package io.sch.historyscan.domain.contexts.codebase.find;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI;

import java.util.List;

@HexagonalArchitectureSPI
@FunctionalInterface
public interface CodeBasesListInventory {
    List<CurrentCodeBase> listAll();
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/find/CodeBasesListInventory.kt
package io.sch.historyscan.domain.contexts.codebase.find

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI

@HexagonalArchitectureSPI
fun interface CodeBasesListInventory {
    fun listAll(): List<CurrentCodeBase>
}
